### PR TITLE
Extract one interface for Pinot Controller API protocol

### PR DESCRIPTION
**Pull request by dongxiaoman**
## Description
The First PR of a series of refactoring Pinot Endpoints, to extract the endpoint interfaces.

It simply extracts all endpoints into an interface, expose it to the outside. Benefits are:
1. These endpoints will become a "standard" and gets much more attention to remain backwards compatible
2. It will be much easier to do unit tests too.

## Alternatives:
It is also possible to use `openapi` and code-gen, which is demonstrated in https://github.com/apache/pinot/pull/8854
but it is too much work and still too much pain. Unless it is a micro-service world, it may be not worth it.

## Tests done
1. Manual verification: started `pinot-quickstart` and start the swagger UI, ran through the endpoint
2. Integration test for sure
3.

### CODE REVIEW

**Comment by dongxiaoman** [comment]
cc @walterddr @apucher for context

File: pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotBrokerRestletResource.java [hunk 1]
```diff
@@ -19,149 +19,88 @@
...
 
-import static org.apache.pinot.spi.utils.CommonConstants.SWAGGER_AUTHORIZATION_KEY;
 
-
-@Api(tags = Constants.BROKER_TAG, authorizations = {@Authorization(value = SWAGGER_AUTHORIZATION_KEY)})
-@SwaggerDefinition(securityDefinition = @SecurityDefinition(apiKeyAuthDefinitions = @ApiKeyAuthDefinition(name =
-    HttpHeaders.AUTHORIZATION, in = ApiKeyAuthDefinition.ApiKeyLocation.HEADER, key = SWAGGER_AUTHORIZATION_KEY)))
 @Path("/")
-public class PinotBrokerRestletResource {
+public class PinotBrokerRestletResource implements PinotBrokerService {
```
Comment (dongxiaoman) [inline pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotBrokerRestletResource.java:42]: For reviewer: this class could be further refactored with delegate pattern. We are not going that far in this PR. One baby step at a time.

**Review by walterddr — COMMENTED** [review]
looks good to me for the refactoring. do we have a centralized doc to describe
- the goal of this refactoring (e.g. what will these interface be used, what variation of the interface impl will be in the future)
- any consolidations or grouping are we planning to do? are we planning to create one Pinot*Service per Pinot*RestletResource?

**Comment by dongxiaoman** [comment]
> looks good to me for the refactoring. do we have a centralized doc to describe
> 
> * the goal of this refactoring (e.g. what will these interface be used, what variation of the interface impl will be in the future)
> * any consolidations or grouping are we planning to do? are we planning to create one Pinot_Service per Pinot_RestletResource?

1. Created a [simple google doc](https://docs.google.com/document/d/1CEHWVOqx5kRzhXd5y1KnWKwUGg6aYss6HGpr_ka0br4/edit?usp=sharing) for long term discussion.
2. Grouping and splitting will be needed, I believe it should be case-by-case for each endpoint. And they can be done in future PRs when all the "extraction" is done first.

**Review by Jackie-Jiang — COMMENTED** [review]
Are we trying to extract the interface for restlet service? If so, I'd suggest naming the interface `PinotBrokerRestletResource` and make the implementation `PinotBrokerRestletResourceImpl`. Currently the interface is already tied with restlet

**Comment by dongxiaoman** [comment]
> 

Yes it is as simple as extracting interface. I am actually trying to use IDE to save some labor.


> Are we trying to extract the interface for restlet service? If so, I'd suggest naming the interface `PinotBrokerRestletResource` and make the implementation `PinotBrokerRestletResourceImpl`. Currently the interface is already tied with restlet

Renamed but the interface `PinotBrokerRestletResource` has to be put into another module; it is because when we load resources by Jersey it was using reflection; interfaces and its implementation cannot be in the same module otherwise Jersey will complain and panic.

**Comment by dongxiaoman** [comment]
@Jackie-Jiang @walterddr test passed. Should be good to go now

File: pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotBrokerRestletResourceImpl.java [hunk 2]
```diff
@@ -19,149 +19,88 @@
...
-import static org.apache.pinot.spi.utils.CommonConstants.SWAGGER_AUTHORIZATION_KEY;
 
-
-@Api(tags = Constants.BROKER_TAG, authorizations = {@Authorization(value = SWAGGER_AUTHORIZATION_KEY)})
-@SwaggerDefinition(securityDefinition = @SecurityDefinition(apiKeyAuthDefinitions = @ApiKeyAuthDefinition(name =
-    HttpHeaders.AUTHORIZATION, in = ApiKeyAuthDefinition.ApiKeyLocation.HEADER, key = SWAGGER_AUTHORIZATION_KEY)))
 @Path("/")
-public class PinotBrokerRestletResource {
-  public static final Logger LOGGER = LoggerFactory.getLogger(PinotBrokerRestletResource.class);
+public class PinotBrokerRestletResourceImpl implements PinotBrokerRestletResource {
```
Comment (Jackie-Jiang) [inline pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotBrokerRestletResourceImpl.java:42]: I see in the other refactor PRs you follow the naming convention of calling the interface `*Service` (`PinotBrokerService` for this resource) and not changing the implementation name. I think we can just go with that to keep them consistent. Sorry for getting back and forth.

**Review by Jackie-Jiang — APPROVED** [review]

File: pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotBrokerRestletResourceImpl.java (hunk 2 above)
Comment (dongxiaoman) [reply pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotBrokerRestletResourceImpl.java]: Thanks for reviewing. Reverted now, should be good to merge.

**Pull request merged**